fastplotlib/graphics/features/_base.py

```diff
@@ -197,13 +203,54 @@ def cleanup_slice(key: Union[int, slice], upper_bound) -> Union[slice, int]:
         stop = upper_bound
 
     elif stop > upper_bound:
-        raise IndexError("Index out of bounds")
+        raise IndexError(f"Index: `{stop}` out of bounds for feature array of size: `{upper_bound}`")
 
     step = key.step
     if step is None:
         step = 1
 
     return slice(start, stop, step)
+    # return slice(int(start), int(stop), int(step))
+
+
+def cleanup_array_slice(key: np.ndarray, upper_bound) -> np.ndarray:
```

Comment:
LGTM, basically just allows a user to pass in a 1D boolean array or integer array for indices right?

Reply:
My only thought is what happens when you pass in a boolean array that is not the same length as your data/colors? Does that mean you will only every index from the beginning to how big your boolean array is?

For example:

graphic = line collection with 25 lines
```
bool_array = [True, True, True, False, False]
graphic.data[bool_array]
```

Would that give you only the first three lines in the collection? Or lines 1-3 and 6-25 as well?

Reply:
Will test out!!

Reply:
yes, so you can do `sine_grahic.colors[ys > 5] = "blue"`